h2o-algos/src/main/java/hex/util/LinearAlgebraUtils.java

```diff
@@ -783,26 +783,50 @@ public static Vec toEigen(Vec src) {
       DKV.put(train);
       created=true;
       Log.info("Reducing the cardinality of a categorical column with " + src.cardinality() + " levels to " + maxLevels);
-      Interaction inter = new Interaction();
-      inter._source_frame = train._key;
-      inter._max_factors = maxLevels; // keep only this many most frequent levels
-      inter._min_occurrence = 2; // but need at least 2 observations for a level to be kept
-      inter._pairwise = false;
-      inter._factor_columns = train.names();
-      train = inter.execImpl(dest).get();
+      train = getInteraction(train._key, train.names(), maxLevels).execImpl(dest).get();
     }
     DataInfo dinfo = new DataInfo(train, null, 0, true /*_use_all_factor_levels*/, DataInfo.TransformType.NONE,
             DataInfo.TransformType.NONE, /* skipMissing */ false, /* imputeMissing */ true,
             /* missingBucket */ false, /* weights */ false, /* offset */ false, /* fold */ false, /* intercept */ false);
     DKV.put(dinfo);
     Gram.GramTask gtsk = new Gram.GramTask(null, dinfo).doAll(dinfo._adaptedFrame);
     // round the numbers to float precision to be more reproducible
-//    double[] rounded = gtsk._gram._diag;
+    // double[] rounded = gtsk._gram._diag;
     double[] rounded = new double[gtsk._gram._diag.length];
     for (int i = 0; i < rounded.length; ++i)
       rounded[i] = (float) gtsk._gram._diag[i];
     dinfo.remove();
-    Vec v = new ProjectOntoEigenVector(multiple(rounded, (int) gtsk._nobs, 1)).doAll(1, (byte) 3, train).outputFrame().anyVec();
```

Comment:
this would make ProjectOntoEigenVector unused and maybe safe to remove

Reply:
it is still used in LinearAlgebraUtils.java#toEigen(), which does similar thing but retruns  Vec v = new ProjectOntoEigenVector( ... ) instead of just double [] array =  multiple( .. ). Array defines encoding and ProjectOntoEigenVector actually encodes the vector by using it.